### DISCUSSION TITLE
Emit isn't received by the originating user if you leave all the rooms

### DISCUSSION THREAD

**Discussion by fireattack** [miguelgrinberg/Flask-SocketIO#2025] [General]
Today I was testing something and find a confusing interaction.

Considering the following code:

```python
@socketio.on('switch_room')
def on_switch_room(data):
    print(data)
    new_room = data['room']
    for r in rooms():
        leave_room(r)
    join_room(new_room)
    emit('change_status', {'message': f'You have joined the new room: {new_room}!'})
```

Apparently, the message sent at the end won't be received by the client side.

According to https://flask-socketio.readthedocs.io/en/latest/api.html#flask_socketio.emit:

> to – Send the message to all the users in the given room, or to the user with the given session ID. If this argument is not set and broadcast is False, then the message is sent only to the originating user.

So I would assume the originating user would always receive it no matter what.

After some debugging, I found that the implementation seems to be that every session is in a hidden room that is named by its session ID. By running `for r in rooms(): leave_room(r)`, you left that hidden room too, So the client can no longer receive any personal emit from server end.

This at least isn't clear to me from the documentation, and I can't seem to find anything about it by some quick googling. 

 I was wondering if we can improve the documentation to make it more clear.

Cheers!

**Comment by miguelgrinberg** [comment]
I think a more appropriate way to address this issue would be to change `rooms()` to not return the user's individual room, the one that is named after the `sid`.
